Gems/PhysX/Code/Source/EditorRigidBodyComponent.h

```diff
@@ -123,7 +122,9 @@ namespace PhysX
 
         Debug::DebugDisplayDataChangedEvent::Handler m_debugDisplayDataChangeHandler;
 
-        EditorRigidBodyConfiguration m_config;
+        EditorRigidBodyConfiguration m_config; //!< Generic properties from AzPhysics.
+        RigidBodyConfiguration
+            m_physxSpecificConfig; //!< Properties specific to PhysX which might not have exact equivalents in other physics engines.
```

Comment:
question: If adding other physics backends becomes a thing, do we expect more specific structs for different backends to be added here? Or would we maybe refactor to have `PhysXEditorRigidBodyComponent` (likey maybe we had before?) and use composition and/or inheritance to than share as much as possible? I know this is a hypothetical but just wanted to check if it'll make this sort of thing more complicated in future?

Reply:
Ah sorry I see this is in the PhysX Gem so I guess if we turned off the PhysX Gem and turned on another then having just `EditorRigidBodyComponent` would be fine so please ignore my earlier comment - it just might be a usability thing to think of in future (do we have the concept of incompatible Gems? 🤔)

Reply:
Yeah, but I was thinking we might be able to move a lot of the generic logic into AzPhysics and have components in backend-specific gems that just extend base functionality from there.